code/client/src/components/AddressInput.jsx

```diff
@@ -24,13 +34,53 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
   const { isMobile } = useWindowDimensions()
 
   const deleteKnownAddress = useCallback((address) => {
-    setAddressCallback({ value: '' })
+    setAddressCallback({ value: '', label: '' })
     dispatch(walletActions.deleteKnownAddress(address))
   }, [dispatch])
 
-  const onSearchAddress = useCallback((address) => {
-    setAddressCallback({ value: address, label: address })
-  }, [setAddressCallback])
+  const onSearchAddress = (value) => {
+    setSearchingAddress(true)
+    setSearchValue(value)
+  }
+
+  useWaitExecution(
+    async () => {
+      try {
+        const validAddress = util.safeNormalizedAddress(searchValue)
+
+        if (validAddress) {
+          const domainName = await api.blockchain.domain.reverseLookup({ address: validAddress })
+
+          setAddressCallback({
+            value: validAddress,
+            domainName,
+            filterValue: searchValue,
+            selected: false
+          })
+        }
+
+        if (!isEmpty(searchValue)) {
+          const resolvedAddress = await api.blockchain.domain.resolve({ name: searchValue })
+
+          if (resolvedAddress && resolvedAddress !== '0x0000000000000000000000000000000000000000') {
+            setAddressCallback({
+              value: resolvedAddress,
+              domainName: searchValue,
+              filterValue: searchValue,
+              selected: false
+            })
+          }
+        }
+
+        setSearchingAddress(false)
+      } catch (e) {
```

Comment:
let's log it to console (which should be also captured by Sentry) so we know if the resolver is temporarily down or if we are blocked due to 429